qa/full-cluster-restart/build.gradle

```diff
@@ -54,11 +54,20 @@ for (Version bwcVersion : bwcVersions.indexCompatible) {
     systemProperty 'tests.is_old_cluster', 'true'
   }
 
+  TaskProvider upgradeCluster = tasks.register("${baseName}#upgradeCluster") {
+    dependsOn "${baseName}#oldClusterTest"
+    doLast {
+      testClusters."${baseName}".goToNextVersion()
+    }
+  }
+
   tasks.register("${baseName}#upgradedClusterTest", RestTestRunnerTask) {
     useCluster testClusters."${baseName}"
-    dependsOn "${baseName}#oldClusterTest"
+    dependsOn upgradeCluster
     doFirst {
-      testClusters."${baseName}".goToNextVersion()
+      if (tasks."${baseName}#oldClusterTest".state.skipped) {
+        throw new GradleException("Cannot execute $name because ${baseName}#oldClusterTest was skipped")
```

Comment:
Could we make this error more actionable e.x. by suggesting `--rerun-tasks` or something like that ? 

I'm assuming there's nothing we can do e.x. when the task-graph is ready to re-configure the skipped task to run ?  Could we maybe at least do the check sooner than `doFirst` ?

Reply:
We cannot. We have no way of knowing if that task will be cached until it actually executes, since input snapshotting happens at that time.

Reply:
> Could we make this error more actionable e.x. by suggesting --rerun-tasks or something like that ?

Done.